### PR TITLE
fix: ensure schema comments are parsed and generated #1489 (part 2)

### PR DESCRIPTION
**Pull request by gak**

### CODE REVIEW

File: go-runtime/compile/build.go [hunk 1]
```diff
@@ -261,17 +260,8 @@ func online() bool {
...
-			commented := "//"
-			// Prevent trailing whitespace on empty lines.
-			if line != "" {
-				commented += " " + line
-			}
-			return commented
-		}), "\n")
-	},
-	"type": genType,
+	"comment": schema.EncodeComments,
```
Comment (alecthomas) [inline go-runtime/compile/build.go:263]: Nice

**Review by alecthomas — APPROVED** [review]

**Review by matt2e — APPROVED** [review]

File: go-runtime/compile/external-module-template/_ftl/go/modules/{{ range .NonMainModules }}{{ push .Name . }}{{ end }}/external_module.go [hunk 2]
```diff
@@ -12,13 +12,13 @@ import (
 var _ = context.Background
 
 {{- range .Decls }}
-{{- if .IsExported}}
-{{if and (is "Enum" .) .IsValueEnum}}
-{{$enumName := .Name -}}
-{{if .Comments -}}
-{{.Comments|comment}}
+{{if .Comments}}
```
Comment (matt2e) [inline go-runtime/compile/external-module-template/_ftl/go/modules/{{ range .NonMainModules }}{{ push .Name . }}{{ end }}/external_module.go:15]: I know @alecthomas just told you to do it this way, but does this introduce a bug where non-exported decls have their comments put in `external_module.go` but they are there dangling without their declaration?

Comment (alecthomas) [reply go-runtime/compile/external-module-template/_ftl/go/modules/{{ range .NonMainModules }}{{ push .Name . }}{{ end }}/external_module.go:15]: If that does happen, we should still do it this way but also check `.IsExported`

Comment (alecthomas) [reply go-runtime/compile/external-module-template/_ftl/go/modules/{{ range .NonMainModules }}{{ push .Name . }}{{ end }}/external_module.go:15]: But I would think if that were the case we would have seen them in the tests...

**Pull request merged**